hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/monitoring/MonitoringService.java

```diff
@@ -0,0 +1,100 @@
+package pl.allegro.tech.hermes.management.infrastructure.monitoring;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.ConsumerGroupDescription;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.common.ConsumerGroupState;
+import org.apache.kafka.common.KafkaFuture;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.common.kafka.ConsumerGroupId;
+import pl.allegro.tech.hermes.common.kafka.KafkaNamesMapper;
+import pl.allegro.tech.hermes.common.kafka.KafkaTopics;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+public class MonitoringService {
+    private static final Logger logger = LoggerFactory.getLogger(MonitoringService.class);
+
+    private final KafkaNamesMapper kafkaNamesMapper;
+    private final AdminClient adminClient;
+    private final String clusterName;
+
+    public MonitoringService(KafkaNamesMapper kafkaNamesMapper, AdminClient adminClient, String clusterName) {
+        this.kafkaNamesMapper = kafkaNamesMapper;
+        this.adminClient = adminClient;
+        this.clusterName = clusterName;
+    }
+
+    public boolean checkIfAllPartitionsAreAssigned(Topic topic, String subscriptionName) {
+        KafkaTopics kafkaTopics = kafkaNamesMapper.toKafkaTopics(topic);
+        ConsumerGroupId consumerGroupId = kafkaNamesMapper.toConsumerGroupId(new SubscriptionName(subscriptionName, topic.getName()));
+        try {
+            return checkIfAllPartitionsAreAssigned(consumerGroupId, kafkaTopics);
+        } catch (Exception e) {
+            logger.error("Failed to describe group with id: {}", consumerGroupId.asString(), e);
+            return true;
+        }
+    }
+
+    private boolean checkIfAllPartitionsAreAssigned(ConsumerGroupId consumerGroupId,
+                                                       KafkaTopics kafkaTopics) throws ExecutionException, InterruptedException {
+        Optional<ConsumerGroupDescription> consumerGroupDescription = getConsumerGroupDescription(consumerGroupId);
+
+        if (isConsumerGroupRebalancing(consumerGroupDescription)) {
+            logger.info("Monitoring. Consumer group is rebalancing: {}", consumerGroupId.asString());
+            return true;
+        }
+
+        int topicPartitions = getTopicPartitions(kafkaTopics);
+        int partitionsInConsumerGroups = getPartitionsInConsumerGroups(consumerGroupDescription);
+
+        if (topicPartitions != partitionsInConsumerGroups) {
+            logger.warn("Monitoring. Cluster {}. Topic and subscription partitions do not match: {}, partitions in consumer groups: {}, "
```

Comment:
Maybe more legible warning would be: `Missing consumer groups for cluster {} and subscription {}. Expected number of consumer groups: {}, but was {}`